lib/tenstorrent/bh_arc/gpio.c

```diff
@@ -8,9 +8,10 @@
 
 #include "reg.h"
 
+#include <tenstorrent/control_reg.h>
```

Comment:
`RESET_UNIT_GPIO_PAD_TRIEN_CNTL_REG_ADDR` is mostly a detail that should be encapsulated inside of the gpio neighbourhood (or possibly pinctrl). I could see maybe naming this file `zephyr/drivers/gpio/gpio_tt_bh.h` to follow the pattern that exists already with [most Zephyr GPIO drivers](https://github.com/zephyrproject-rtos/zephyr/tree/main/include/zephyr/drivers/gpio).

For reference #265

 Is this register location specific to Blackhole or all Tenstorrent products?